Remotion/Data/DomainObjects/Persistence/Rdbms/StorageProviderCommands/DelegateBasedReadOnlyCommand.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms.StorageProviderCommands;
+
+/// <summary>
+/// The <see cref="DelegateBasedReadOnlyCommand{TIn,TOut}"/> executes an <see cref="IRdbmsProviderReadOnlyCommand{T}"/>
+/// and applies a specified operation-transformation to the result.
+/// </summary>
+public class DelegateBasedReadOnlyCommand<TIn, TOut> : IRdbmsProviderReadOnlyCommand<TOut>
+{
+  private readonly IRdbmsProviderReadOnlyCommand<TIn> _command;
+  private readonly Func<TIn, TOut> _operation;
+
+  public DelegateBasedReadOnlyCommand (IRdbmsProviderReadOnlyCommand<TIn> command, Func<TIn, TOut> operation)
+  {
+    ArgumentUtility.CheckNotNull("command", command);
+    ArgumentUtility.CheckNotNull("operation", operation);
+
+    _command = command;
+    _operation = operation;
+  }
+
+  public IRdbmsProviderReadOnlyCommand<TIn> Command
+  {
+    get { return _command; }
+  }
+
+  public Func<TIn, TOut> Operation
+  {
+    get { return _operation; }
+  }
+
+  public TOut Execute (IRdbmsProviderReadWriteCommandExecutionContext executionContext)
```

Comment:
@MichaelKetting I don't understand. Do you mean "make this an explicit interface implementation"?

Reply:
Since this is the ReadOnly-specific DelegateCommand, it feels "correct" if the readwrite Execute overload is implemented as an explicit interface implementation isntead of implicit. Not sure if that holds up, now that we have settled on new names, though.